### PR TITLE
bump golangci

### PR DESCRIPTION
**Pull request by honza**
The previous version doesn't work with golang 1.18.  The latest version
seems to break rule G109:

    Potential Integer overflow made by strconv.Atoi result conversion to
    int16/32

It's broken in that it doesn't respect our #nosec directive.  I disabled
that rule in .golangci.yaml.

We also update go.mod, and vendor.

cc #256

### CODE REVIEW

**Comment by sosiouxme** [comment]
I can't easily tell if it's already doing this - can you test what happens against go 1.18 or does that need to be a second PR?

File: .golangci.yml [hunk 1]
```diff
@@ -36,3 +36,6 @@ issues:
    - linters:
      - staticcheck
      text: "SA1019:"
+   - linters:
```
Comment (dtantsur) [inline .golangci.yml:39]: Could you leave a FIXME to remove it once goling is fixed?

**Comment by honza** [comment]
> I can't easily tell if it's already doing this - can you test what happens against go 1.18 or does that need to be a second PR?

I just opened #259 that combines the two to see if this works.

@sosiouxme

**Pull request merged**